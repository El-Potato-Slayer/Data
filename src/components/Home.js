import { Link } from 'react-router-dom';

function Home() {
  return (
    <>
      <section className="header">
        <h1>Tactus Caelesti</h1>
      </section>
      <div className="sub-header">
        <Link to="/planets">
          <section>
            <h2>Planets</h2>
          </section>
        </Link>
        <Link to="/blackholes">
          <section>
            <h2>Moons</h2>
          </section>
        </Link>
      </div>
    </>
  );
}

export default Home;
