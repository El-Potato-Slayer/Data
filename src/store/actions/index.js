const setBodies = (payload) => ({
  type: 'SET_BODIES',
  payload,
});

export default setBodies;
