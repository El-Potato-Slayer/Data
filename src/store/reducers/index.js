import { combineReducers } from 'redux';
import solarSystemReducer from './solarSystemReducer';
import filterReducer from './filterReducer';

export default combineReducers({ solarSystemReducer, filterReducer });
