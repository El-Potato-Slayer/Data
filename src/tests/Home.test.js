import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import store from '../store/configureStore';
import '@testing-library/jest-dom';
import Home from '../components/Home';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>, div,
  );
});

it('renders correctly', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/']}>
        <Home />
      </MemoryRouter>
    </Provider>,
  );
  expect(getByTestId('home-header')).toHaveTextContent('Tactus');
});

it('renders incorrectly', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/']}>
        <Home />
      </MemoryRouter>
    </Provider>,
  );
  expect(getByTestId('home-header')).not.toHaveClass('Thanos');
});

it('renders snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
