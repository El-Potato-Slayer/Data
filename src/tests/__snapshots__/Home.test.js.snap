// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
Array [
  <a
    className="header"
    data-testid="home-header"
    href="/all"
    onClick={[Function]}
  >
    <h1>
      Tactus Caelesti
    </h1>
  </a>,
  <div
    className="sub-header"
  >
    <a
      href="/planets"
      onClick={[Function]}
    >
      <section>
        <h2>
          Planets
        </h2>
      </section>
    </a>
    <a
      href="/blackholes"
      onClick={[Function]}
    >
      <section>
        <h2>
          Blackholes
        </h2>
      </section>
    </a>
  </div>,
]
`;
